### PR TITLE
doc: Update Istio Getting Started Guide to Istio 0.6.0

### PR DESCRIPTION
**Pull request by rlenglet**
Signed-off-by: Romain Lenglet <romain@covalent.io>

### CODE REVIEW

**Comment by rlenglet** [comment]
Validated that the GSG works with images based on https://github.com/cilium/cilium/pull/3035. These GSG changes can be merged now before https://github.com/cilium/cilium/pull/3035 is merged.

**Comment by rlenglet** [comment]
test-me-please

File: Documentation/gettingstarted/istio.rst [hunk 1]
```diff
@@ -93,15 +99,30 @@ Deploy Istio on Kubernetes:
 
     $ kubectl create -f ${ISTIO_HOME}/install/kubernetes/istio.yaml
 
-Check the progress of the deployment (every services should have an
+.. TODO:
+   Rewrite istio.yaml to replace the proxy and proxy_init images with
+   Cilium's images.
+
+Configure Istio's sidecar injection to use Cilium's Docker images for the
```
Comment (tgraf) [inline Documentation/gettingstarted/istio.rst:106]: This needs an explanation that we a replacing the images in order to allow for Istio and Cilium to share the Envoy proxy.

**Review by tgraf — CHANGES_REQUESTED** [review]
Looks great. Needs a intro section which explains the intent of the guide.

File: Documentation/gettingstarted/istio.rst (hunk 1 above)
Comment (rlenglet) [reply Documentation/gettingstarted/istio.rst:106]: That’s not yet true. I was thinking of writing this up when we’re really running the L7 filter in the sidecar.

Comment (tgraf) [reply Documentation/gettingstarted/istio.rst:106]: OK, makes sense

**Review by tgraf — APPROVED** [review]

**Pull request merged**